tests/test_numeric.py

```diff
@@ -415,3 +415,18 @@ def test_matrix_2dindex(self):
     def test_matrix_slice(self):
         A = numeric.accumulate(numpy.array([[1, 2], [3, 4]]), (numpy.array([0, 1]), slice(0, 4, 3)), (2,4))
         self.assertEqual(A.tolist(), [[1, 0, 0, 2], [3, 0, 0, 4]])
+
+
+class compress_indices(TestCase):
+
+    def test(self):
+        indices = numpy.array([0, 2, 2, 2, 5, 6, 7])
+        length = 8
+        c = numeric.compress_indices(indices, length)
+        for n in range(length):
```

Comment:
Why not just compute the desired value by hand? This unittest doesn't test the length of `c` (a `c` that is too long goes unnoticed).

Reply:
I think unit tests should check for properties of the return value where possible, rather than compare against precomputed values that cannot be easily verified. I added the length check though.